MV_APPEND tests failing on geo_shape

### CI Link

local

### Repro line

./gradlew ':x-pack:plugin:esql:test' --tests "org.elasticsearch.xpack.esql.expression.function.scalar.multivalue.MvAppendTests.testEvaluateBlockWithoutNulls {TestCase=<geo_shape>, <geo_shape>}" -Dtests.seed=D9F77BE783139D12 -Dtests.locale=es-SV -Dtests.timezone=Europe/Riga -Druntime.java=22

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

```
  1> [2024-06-05T01:09:31,200][INFO ][o.e.x.e.e.f.s.m.MvAppendTests] [testEvaluateBlockWithoutNulls] [TestCase=<geo_shape>, <geo_shape>] after test
  2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:esql:test' --tests "org.elasticsearch.xpack.esql.expression.function.scalar.multivalue.MvAppendTests.testEvaluateBlockWithoutNulls {TestCase=<geo_shape>, <geo_shape>}" -Dtests.seed=D9F77BE783139D12 -Dtests.locale=es-SV -Dtests.timezone=Europe/Riga -Druntime.java=22
  2> java.lang.AssertionError: Test data is too large to fit in the memory
        at __randomizedtesting.SeedInfo.seed([D9F77BE783139D12:93E9D00A2A5B0AA6]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.assertTrue(Assert.java:42)
        at org.junit.Assert.assertFalse(Assert.java:65)
        at org.elasticsearch.xpack.esql.expression.function.AbstractFunctionTestCase.testEvaluateBlockWithoutNulls(AbstractFunctionTestCase.java:313)
        at java.base/jdk.internal.reflect.DirectMethodHandl
```

I think we need to limit the size of the shape a bit.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -53,9 +53,6 @@ tests:
 - class: "org.elasticsearch.client.RestClientSingleHostIntegTests"
   issue: "https://github.com/elastic/elasticsearch/issues/102717"
   method: "testRequestResetAndAbort"
-- class: org.elasticsearch.xpack.esql.expression.function.scalar.multivalue.MvAppendTests
-  issue: https://github.com/elastic/elasticsearch/issues/109373
-  method: testEvaluateBlockWithoutNulls
 - class: "org.elasticsearch.xpack.deprecation.DeprecationHttpIT"
   issue: "https://github.com/elastic/elasticsearch/issues/108628"
   method: "testDeprecatedSettingsReturnWarnings"

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvAppendTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvAppendTests.java
@@ -235,8 +235,8 @@ public class MvAppendTests extends AbstractFunctionTestCase {
         }));
 
         suppliers.add(new TestCaseSupplier(List.of(DataType.GEO_SHAPE, DataType.GEO_SHAPE), () -> {
-            List<Object> field1 = randomList(1, 10, () -> new BytesRef(GEO.asWkt(GeometryTestUtils.randomGeometry(randomBoolean()))));
-            List<Object> field2 = randomList(1, 10, () -> new BytesRef(GEO.asWkt(GeometryTestUtils.randomGeometry(randomBoolean()))));
+            List<Object> field1 = randomList(1, 5, () -> new BytesRef(GEO.asWkt(GeometryTestUtils.randomGeometry(randomBoolean()))));
+            List<Object> field2 = randomList(1, 5, () -> new BytesRef(GEO.asWkt(GeometryTestUtils.randomGeometry(randomBoolean()))));
             var result = new ArrayList<>(field1);
             result.addAll(field2);
             return new TestCaseSupplier.TestCase(
@@ -251,8 +251,8 @@ public class MvAppendTests extends AbstractFunctionTestCase {
         }));
 
         suppliers.add(new TestCaseSupplier(List.of(DataType.CARTESIAN_SHAPE, DataType.CARTESIAN_SHAPE), () -> {
-            List<Object> field1 = randomList(1, 10, () -> new BytesRef(CARTESIAN.asWkt(ShapeTestUtils.randomGeometry(randomBoolean()))));
-            List<Object> field2 = randomList(1, 10, () -> new BytesRef(CARTESIAN.asWkt(ShapeTestUtils.randomGeometry(randomBoolean()))));
+            List<Object> field1 = randomList(1, 5, () -> new BytesRef(CARTESIAN.asWkt(ShapeTestUtils.randomGeometry(randomBoolean()))));
+            List<Object> field2 = randomList(1, 5, () -> new BytesRef(CARTESIAN.asWkt(ShapeTestUtils.randomGeometry(randomBoolean()))));
             var result = new ArrayList<>(field1);
             result.addAll(field2);
             return new TestCaseSupplier.TestCase(

